core/controlplane/config/templates/cluster.yaml

```diff
@@ -185,7 +185,16 @@ kmsKeyArn: "{{.KMSKeyARN}}"
 #  # ATTENTION: Consider limiting number of characters in clusterName and managedIamRoleName to avoid the resulting IAM
 #  # role name's length from exceeding the AWS limit: 64
 #  # See https://github.com/kubernetes-incubator/kube-aws/issues/347
-#  managedIamRoleName: "yourManagedRole"
+#  # if you omit this block kube-aws would create an IAM Role and a customer managed policy with a random name.
+#  #iam:
+#       # This will create a named IAM Role managed by kube-aws with the name {AWS::Region}-$managedIamRoleName.
+#       # It will have attached a customer Managed Policy that you can modify afterwards if you need more permissions for your cluster.
+#       # Be careful with the Statements you modify because an update could overwrite your own.
+#       # the Statements included in the ManagedPolicy are the minimun ones required for the Controllers to run.
+#       # managedIamRoleName: "yourManagedRole"
+#       # ATTENTION: if you decide to use an existing InstanceProfile you need to ensure it has the needed permissions for kube-aws to run.
+#       # kube-aws will not manage and never manage this InstanceProfile. IAM Roles will not be created by kube-aws in this case.
+#       # existingInstanceProfile: "arn:aws:iam::YOURACCOUNTID:instance-profile/INSTANCEPROFILEID"
```

Comment:
I guess `iam` should be `iamInstanceProfile` so that it is even more clearer that it is used to customize `IamInstanceProfile` associated to nodes.

If we go that way, `iam.existingInstanceProfile` could be just `iamInstanceProfile.arn`. It will correspond more to e.g. `subnets[].id` or `apiEndpoints[].loadBalancer.id` which is there for reusing existing AWS resources.